deploy/modules/gcp/tidb-cluster/main.tf

```diff
@@ -138,7 +142,7 @@ module "tidb-cluster" {
   override_values            = var.override_values
   kubeconfig_filename        = var.kubeconfig_path
   base_values                = file("${path.module}/values/default.yaml")
-  wait_on_resource           = [google_container_node_pool.tidb_pool]
+  wait_on_resource           = [var.tidb_operator_id]
```

Comment:
tidb cluster resource must be applied after tidb-operator is ready

Reply:
The reason I used `tidb_pool` for `wait_on_resource` was because the console output during apply was pretty noisy until the tidb pods were ready. It would be nice to keep that, but if it is too difficult then no worries. WDYT?

Reply:
I see. I made this change because node pools and tidb-operator are applied in parallel. When `tidb_pool` is ready, tidb-operator may not be installed successfully.

```
- [tidb-operator] cluster -> setup-env -> pingcap repo -> tidb-operator helm chart
- [tidb-cluster]                 -> tikv_pool/monitor_pool/pd_pool/tidb_pool
```
 
`tidb-cluster` can wait for both tidb-operator (CRD, etc) and tidb_pool (node resources). I'll update this PR.

Reply:
Updated in https://github.com/pingcap/tidb-operator/pull/819/commits/a1d2cb73fde4f60502ee416ef7e8f7f9f5e873f4.